src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,55 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int DOUBLE_ARRAY_SIZE = 20;
+    private static final int ARRAY_SIZE = 10;
+    private Object[] array = new Object[DOUBLE_ARRAY_SIZE];
+    private int arrayIterator = 0;
+    private int sizeCounter = 0;
+
     @Override
     public void put(K key, V value) {
+        if (get(key) != null) {
+            int currentArraySize = sizeCounter * 2 - 1;
+            for (int i = 0; i < currentArraySize; i++) {
+                if (i % 2 == 0) {
+                    if (key == null && array[i] == null) {
+                        array[i + 1] = value;
+                        break;
+                    }
+                    if (key != null && key.equals(array[i])) {
+                        array[i + 1] = value;
+                        break;
+                    }
```

Comment:
There is no need in 2 if blocks. We may perform null-safe comparison within one
```suggestion
                    if (key == array[i] || key != null && key.equals(array[i])) {
                        array[i + 1] = value;
                        break;
                    }
```